PerfectWedding/src/main/java/com/infoshareacademy/Utils.java

```diff
@@ -0,0 +1,239 @@
+                    }
+                }
+            } catch (Exception e) {
+                System.out.println("nieodpowiedia wartosc.");
+                toReturn=scanForInt(prompt, min, max, singleSelection);
+            }
+//        } while (toReturn.size() == 0);
+        return toReturn;
+    }
+
+    public static String scanForString(String prompt) {
+        System.out.println(prompt);
+        String toReturn = "";
+//        do {
+            try {
+                Scanner scanner = new Scanner(System.in);
+                toReturn = scanner.nextLine();
+            } catch (Exception e) {
+                System.out.println("nieodpowiedia wartosc.");
+                toReturn=scanForString(prompt);
+            }
+//        } while (Objects.equals(toReturn, ""));
+        return toReturn;
+    }
+
+    public static String scanForString(String prompt, String... allowedAnswers) {
+        String toReturn = "";
+//        do {
+            System.out.println(prompt);
+            System.out.println("Dopuszczalne wartosci to: ");
+            System.out.println(Arrays.toString(allowedAnswers));
+            try {
+                Scanner scanner = new Scanner(System.in);
+                toReturn = scanner.nextLine();
+                boolean isAnswerCorrect = false;
+                for (int i = 0; i < allowedAnswers.length; i++) {
+                    if (toReturn.equals(allowedAnswers[i])) {
+                        isAnswerCorrect = true;
+                        break;
+                    }
+                }
+                if (!isAnswerCorrect) {
+                    throw new Exception("not in the list of allowed answers");
+                }
+            } catch (Exception e) {
+                System.out.println("nieodpowiedia wartosc.");
+                toReturn = scanForString(prompt, allowedAnswers);
+            }
+//        } while (toReturn.equals(""));
+        return toReturn;
+    }
+
+    public static <T> String arrayToString(T[] array, Boolean printSequenceNumbers) {
+        String toReturn = "";
+        for (int i = 0; i < array.length; i++) {
+            if (printSequenceNumbers) {
+                toReturn = toReturn + (i + 1) + "." + array[i] + "\n";
+            } else {
+                toReturn = toReturn + array[i] + "\n";
+            }
+        }
+        return toReturn;
+    }
+
+    public static <T> String listToString(List<T> list, Boolean printSequenceNumbers) {
+        String toReturn = "";
+        for (int i = 0; i < list.size(); i++) {
+            if (printSequenceNumbers) {
+                toReturn = toReturn + (i + 1) + "." + list.get(i) + "\n";
+            } else {
+                toReturn = toReturn + list.get(i) + "\n";
+            }
+        }
+        return toReturn;
+    }
+    public static void setVoivodeship(ServiceProvider serviceProvider) {
+        serviceProvider.setLocation(new Location(scanInput("Podaj lokalizcje firmy (miasto)")));
```

Comment:
Ja bym zmienił nazwę metody lub w ogóle wyciął z niej fragment ustalający miejscowość.
W ProvidersEdit już ustalamy miejscowość i jest to zdublowane.

Można pójść dwoma ścieżkami - w tej metodzie to zostawić, ale zmienić (rozszerzyć) jej nazwę lub usunąć ustalanie miejscowości z tej metody.